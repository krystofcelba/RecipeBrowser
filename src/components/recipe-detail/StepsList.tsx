import React from 'react';

import i18n from 'src/assets/i18n';
import { View, SubTitleText, LeftPaddedBoldText, LeftPaddedText } from 'src/components/common';

interface Props {
  steps: [string];
}

const StepsList = ({ steps }: Props) => (
  <View flex={1}>
    <SubTitleText>{i18n.t('steps')}</SubTitleText>
    {steps.map((step, i) => (
      <View flexDirection="row" key={i}>
        <LeftPaddedBoldText>{`${1 + i}.`}</LeftPaddedBoldText>
        <LeftPaddedText>{step}</LeftPaddedText>
      </View>
    ))}
  </View>
);

export default StepsList;
