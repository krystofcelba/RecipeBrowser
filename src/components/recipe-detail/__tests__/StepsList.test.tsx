import React from 'react';
import { shallow } from 'enzyme';
import StepsList from '../StepsList';

const mockedRecipes = require('src/__mockData__/recipes.json');

describe('StepsList component', () => {
  const mockedRecipe = mockedRecipes[0];
  it('renders as expected', () => {
    const wrapper = shallow(<StepsList steps={mockedRecipe.steps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
