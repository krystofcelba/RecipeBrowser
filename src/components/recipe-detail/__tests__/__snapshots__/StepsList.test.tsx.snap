// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsList component renders as expected 1`] = `
<glamorous.View
  flex={1}
>
  <SubTitleText>
    Steps
  </SubTitleText>
  <glamorous.View
    flexDirection="row"
    key="0"
  >
    <LeftPaddedBoldText>
      1.
    </LeftPaddedBoldText>
    <LeftPaddedText>
      Halve eggs and remove yolk.
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="1"
  >
    <LeftPaddedBoldText>
      2.
    </LeftPaddedBoldText>
    <LeftPaddedText>
      Mix yolks with mustard, cottage cheese and half of dill.
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="2"
  >
    <LeftPaddedBoldText>
      3.
    </LeftPaddedBoldText>
    <LeftPaddedText>
      Fill egg halves with yolk mixture and garnish with remaining dill.
    </LeftPaddedText>
  </glamorous.View>
</glamorous.View>
`;
